src/Streamly/Fold.hs

```diff
@@ -1665,16 +1694,18 @@ splitSuffixBy' predicate f m =
 --
 -- > splitOn . intercalate == id
 --
--- The following law always holds:
---
--- > concat . splitOn . intercalate == concat
---
 -- @since 0.7.0
-{-# INLINE splitOn #-}
-splitOn
+
+-- XXX We can use a polymorphic vector implemented by Array# to represent the
+-- sequence, that way we can avoid the Storable constraint. If we still need
+-- Storable Array for performance, we can use a separate splitOnArray API for
+-- that. We can also have an API where the sequence itself is a lazy stream, so
+-- that we can search files in files for example.
+{-# INLINE splitOnSeq #-}
+splitOnSeq
     :: (IsStream t, MonadIO m, Storable a, Enum a, Eq a)
     => Array a -> Fold m a b -> t m a -> t m b
-splitOn patt f m = D.fromStreamD $ D.splitOn patt f (D.toStreamD m)
+splitOnSeq patt f m = D.fromStreamD $ D.splitOn patt f (D.toStreamD m)
```

Comment:
Don't think this is a good name, how about using full sequence instead, `splitOnSequence`

Reply:
It becomes a little longer. The problem may not be that this becomes longer, but the problem is that if we introduce its cousins like `splitOnSuffixSequence` later they become even longer. I am still willing to consider it if the `Seq` prefix sounds less intuitive/confusing, its just trade off with longer name vs more explicit.

Reply:
Does using a `Splitter` record like the split package, as mentioned in one comment, do away with the long names?

Reply:
I am not sure if we will ever implement something like that, it may not be worth the complexity. Just using prefix, infix, suffix style splitting covers a lot of ground, and we may not even want these as we will have other ways, like parser combinators, to achieve custom splitting.